nixos/modules/services/hardware/udev.nix

```diff
@@ -171,6 +171,11 @@ let
       mv etc/udev/hwdb.bin $out
     '';
 
+  compressFirmware = if config.boot.kernelPackages.kernelAtLeast "5.3" then
```

Comment:
It probably would been better to check the actual kernel options required for this, instead of just the kernel version. E.g. for downstream users with custom kernel configs.

Also maybe this should have been made an option.

Which isn't to say that I don't think this is a cool change and I like it, just that it lead to me having to spend some time trying to figure out why the wifi on my raspberry pis (with a custom kernel (config)) doesn't work anymore.

Reply:
I'm also running into this issue on my raspberry pi. Is there support in NixOS to check for kernel config options in a place like this?

Reply:
There is support to check that, yes. Although I couldn't even get it to work even after setting all the options that I thought need setting, so I just ended up overlaying `compressFirmwareXz` to `lib.id` for these systems.

Reply:
Thanks! I was looking for an overlay fix and this works :+1:

Reply:
> It requires an IFD (import from derivation) as the configuration is _built_ using a Nix build. This in turn causes other issues.

Yes and no. That's sort of already how all of this works, right now. We _have_ `system.requiredKernelConfig`.
`manual-config.nix` kernel already has a `allowImportFromDerivation` parameter (which defaults to false).

IMHO our kernel config and build system, which features [amazing code like this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/generate-config.pl#L122-L126), is in need of an overhaul.